packages/@react-stately/combobox/src/useComboBoxState.ts

```diff
@@ -79,27 +84,56 @@ export function useComboBoxState<T extends object>(props: ComboBoxStateProps<T>)
     items: props.items ?? props.defaultItems
   });
 
+  // This is only used if items are not controlled
+  let originalCollection = collection;
+
   let filteredCollection = useMemo(() => (
     // No default filter if items are controlled.
     props.items != null || !defaultFilter
       ? collection
       : filterCollection(collection, inputValue, defaultFilter)
   ), [collection, inputValue, defaultFilter, props.items]);
 
-  let triggerState = useMenuTriggerState(props);
-  let open = (focusStrategy?: FocusStrategy) => {
+  // Track what action is attempting to open the menu
+  let menuOpenTrigger = useRef('focus' as MenuTriggerAction);
+  let onOpenChange = (open: boolean) => {
+    if (props.onOpenChange) {
+      props.onOpenChange(open, open ? menuOpenTrigger.current : undefined);
+    }
+  };
+
+  let triggerState = useMenuTriggerState({...props, onOpenChange});
+  let open = (focusStrategy?: FocusStrategy, showAllItems?: boolean, trigger?: MenuTriggerAction) => {
```

Comment:
hmm not sure I follow. you mean if someone called `comboBoxState.open()` manually?

Reply:
yeah, or if someone wrote their own `useComboBox` but used our `useComboBoxState` and wanted their combobox not to show all items when opening the menu. They could then have something like this in their custom `useComboBox` hook:
```
let onKeyDown = (e: KeyboardEvent) => {
    switch (e.key) {
      case 'ArrowDown':
        state.open('first', false, 'manual');
```
meaning their default combobox behavior would be to only show the filtered collection at all times

Reply:
Hmm, I think this may be providing too much flexibility. Seems like in that case maybe it would be preferable to add a prop to control the behavior for everyone equally. I guess if we decide to do that it would also solve this. For now, I think keeping the API simple is preferable.